src/os/FileJournal.cc

```diff
@@ -1576,7 +1559,63 @@ void FileJournal::check_aio_completion()
 }
 #endif
 
-void FileJournal::submit_entry(uint64_t seq, bufferlist& e, int alignment,
+int FileJournal::_op_journal_transactions_prepare(list<ObjectStore::Transaction*>& tls, bufferlist& tbl) {
+  dout(10) << "_op_journal_transactions_prepare " << tls << dendl;
+  unsigned data_len = 0;
+  int data_align = -1; // -1 indicates that we don't care about the alignment
+  for (list<ObjectStore::Transaction*>::iterator p = tls.begin();
+      p != tls.end(); ++p) {
+    ObjectStore::Transaction *t = *p;
+    if (t->get_data_length() > data_len &&
+     (int)t->get_data_length() >= g_conf->journal_align_min_size) {
+     data_len = t->get_data_length();
+     data_align = (t->get_data_alignment() - tbl.length()) & ~CEPH_PAGE_MASK;
+    }
+    ::encode(*t, tbl);
+  }
+  // add it this entry
+  entry_header_t h;
+  unsigned head_size = sizeof(entry_header_t);
+  off64_t base_size = 2*head_size + tbl.length();
+  memset(&h, 0, sizeof(h));
+  if (data_align >= 0)
+    h.pre_pad = ((unsigned int)data_align - (unsigned int)head_size) & ~CEPH_PAGE_MASK;
+  off64_t size = ROUND_UP_TO(base_size + h.pre_pad, header.alignment);
+  unsigned post_pad = size - base_size - h.pre_pad;
+  h.len = tbl.length();
+  h.post_pad = post_pad;
+  h.crc32c = tbl.crc32c(0);
+  dout(10) << " len " << tbl.length() << " -> " << size
+       << " (head " << head_size << " pre_pad " << h.pre_pad
+       << " ebl " << tbl.length() << " post_pad " << post_pad << " tail " << head_size << ")"
+       << " (ebl alignment " << data_align << ")"
+       << dendl;
+  bufferptr ptr = buffer::create_aligned(size, CEPH_MINIMUM_BLOCK_SIZE);
```

Comment:
@liewegas , 
1. buffer::create_aligned(size, CEPH_MINIMUM_BLOCK_SIZE) would malloc  specified bytes memory with CEPH_MINIMUM_BLOCK_SIZE align, is I miss some critical?